orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/image/ImageTaggerSpec.groovy

```diff
@@ -35,33 +35,41 @@ abstract class ImageTaggerSpec<T extends ImageTagger> extends Specification {
   @Unroll
   def "should extract imageId from upstream stages"() {
     given:
-    def pipeline = Execution.newPipeline("orca")
-    def stage1 = new Stage(pipeline, "stage1", stage1Context + [cloudProvider: cloudProvider])
-    def stage2 = new Stage(pipeline, "stage2", stage2Context + [cloudProviderType: cloudProvider])
-    def stage3 = new Stage(pipeline, "stage3", [:])
+    def pipeline = new Execution(Execution.ExecutionType.PIPELINE, "orca")
+    def stage1 = new Stage(pipeline, "bake", "stage1", stage1Context + [cloudProvider: cloudProvider])
+    def stage2 = new Stage(pipeline, "findImageFromTags", "stage2", stage2Context + [cloudProviderType: cloudProvider])
+    def stage3 = new Stage(pipeline, "bake", "stage3", [:])
 
-    stage3.requisiteStageRefIds = [stage2.id.toString()]
-    stage2.requisiteStageRefIds = [stage1.id.toString()]
+    stage3.requisiteStageRefIds = [stage2.name]
+    stage2.requisiteStageRefIds = [stage1.name]
```

Comment:
Can we change these assignments to `[stageX.refId]`? Although technically allowed, it's a little confusing to have the `name` and `refId` be the same value, so maybe move these two assignments below the next block where the `refId`s are being added to the stages.

Reply:
Fixed